server/hwapi/endpoints/certification/logic.py

```diff
@@ -46,27 +51,203 @@ def is_certified(system_info: CertificationStatusRequest, db: Session):
+            model=device.name,
+            vendor=device.vendor.name,
+            capacity=info.capacity,
+        ),
+    )
+
+    wireless_devices = _get_matching_devices(
+        db,
+        devices_query,
+        system_info.wireless or [],
+        repository.find_matching_wireless_device,
+        data_validators.WirelessAdapterValidator,
+        lambda device, _: data_validators.WirelessAdapterValidator(
+            model=device.name, vendor=device.vendor.name, identifier=device.identifier
+        ),
+    )
+
+    audio_devices = _get_matching_devices(
+        db,
+        devices_query,
+        system_info.audio or [],
+        repository.find_matching_audio_device,
+        data_validators.AudioValidator,
+        lambda device, _: data_validators.AudioValidator(
+            model=device.name, vendor=device.vendor.name, identifier=device.identifier
+        ),
+    )
+
+    capture_devices = _get_matching_devices(
+        db,
+        devices_query,
+        system_info.video or [],
+        repository.find_matching_capture_device,
+        data_validators.VideoCaptureValidator,
+        lambda device, _: data_validators.VideoCaptureValidator(
+            model=device.name, vendor=device.vendor.name, identifier=device.identifier
+        ),
+    )
+
+    pci_devices = _get_matching_devices(
+        db,
+        devices_query,
+        system_info.pci_peripherals or [],
+        repository.find_matching_pci_device,
+        data_validators.PCIPeripheralValidator,
+        lambda device, _: data_validators.PCIPeripheralValidator(
+            name=device.name, vendor=device.vendor.name, pci_id=device.identifier
+        ),
+    )
+
+    usb_devices = _get_matching_devices(
+        db,
+        devices_query,
+        system_info.usb_peripherals or [],
+        repository.find_matching_usb_device,
+        data_validators.USBPeripheralValidator,
+        lambda device, _: data_validators.USBPeripheralValidator(
+            name=device.name, vendor=device.vendor.name, usb_id=device.identifier
+        ),
+    )
+
+    # If there are no matching devices of any type, return False
+    if not any(
+        [
+            gpus,
+            cpus,
+            network_devices,
+            wireless_devices,
+            audio_devices,
+            capture_devices,
+            pci_devices,
+            usb_devices,
+        ]
+    ):
+        return False, None
+
+    return True, RelatedCertifiedSystemExistsResponse(
+        architecture=system_info.architecture,
+        board=(
+            data_validators.BoardValidator(
+                manufacturer=board.vendor.name,
+                version=board.version,
+                product_name=board.name,
+            )
+            if board
+            else None
+        ),
+        processor=cpus,
+        gpu=gpus,
+        audio=audio_devices,
+        video=capture_devices,
+        network=network_devices,
+        wireless=wireless_devices,
```

Comment:
The mapping between singular and plural is a bit weird. Would be great if we stick to one

Also it would be nice to have consistent vocabulary for instance either stick with `processor` or `cpu` and `video` or `capture_device`